src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/IHostingEnvironment.cs

```diff
@@ -26,7 +26,7 @@ public interface IHostingEnvironment
         /// Gets or sets the name of the application. This property is automatically set by the host to the assembly containing
         /// the application entry point.
         /// </summary>
-        string ApplicationName { get; set; }
+        string? ApplicationName { get; set; }
```

Comment:
@eerhardt This one seems odd to be nullable, similar to EnvironmentName above, this is set automatically by the host

Reply:
Given the current code, this annotation is correct as there are scenarios where it can be null:

https://github.com/dotnet/runtime/blob/05cb7f5a6eefbcd0fbab3bb5924c07b188d12501/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L202-L213